docs/user_guide/source/engines/inline.rst

```diff
@@ -6,21 +6,50 @@ The Inline engine provides in-process communication between the writer and reade
 
 .. code-block:: c++
 
     adios2::IO inlineIO = adios.DeclareIO("ioName");
     inlineIO.SetEngine("Inline");
+    inlineIO.SetParameters({{"writerID", "inline_write"}, {"readerID", "inline_read"}});
     adios2::Engine inlineWriter = inlineIO.Open("inline_write", adios2::Mode::Write);
-    io.SetParameters({{"writerID", "inline_write"}});
     adios2::Engine inlineReader = inlineIO.Open("inline_read", adios2::Mode::Read);
 
-Notice that unlike other engines, the reader and writer share an IO instance. Also, the ``writerID`` parameter allows the reader to connect to the writer.
+Notice that unlike other engines, the reader and writer share an IO instance. Also, the ``writerID`` parameter allows the reader to connect to the writer, and ``readerID`` allows writer to connect to the reader.
 
-For successful operation, the writer will perform a step, then the reader will perform a step in the same process. Data is decomposed between processes, and the writer can write its portion of the data like other Adios engines. When the reader starts its step, the only data it has available is that written by the writer in its process. To select this data in Adios, use a block selection. The reader then can retrieve whatever data was written by the writer.
+For successful operation, the writer will perform a step, then the reader will perform a step in the same process. Data is decomposed between processes, and the writer can write its portion of the data like other ADIOS engines. When the reader starts its step, the only data it has available is that written by the writer in its process. To select this data in ADIOS, use a block selection. The reader then can retrieve whatever data was written by the writer.
 
-Please refer to the examples or tests for the typical access pattern. Note, however, that the ``Get()`` method:
+.. note::
+ The inline engine does not support Sync mode for writing. In addition, since the inline engine does not do any data copy, the writer should avoid changing the data contents before the reader has read the data.
+
+Typical access pattern:
+
+.. code-block:: c++
+ // ... Application data generation
+
+ inlineWriter.BeginStep();
+ inlineWriter.Put(var, data); // always use deferred mode
+ inlineWriter.EndStep();
+ // Unlike other engines, data should not be reused at this point,
+ // though ADIOS cannot enforce this.
+ // Must wait until reader is finished using the data.
+
+ inlineReader.BeginStep();
+ auto blocksInfo = inlineReader.BlocksInfo(var, step);
+ for (auto& info : blocksInfo)
+ {
+     var.SetBlockSelection(info.BlockID);
+     inlineReader.Get(var, info);
+ }
+ inlineReader.EndStep();
+
+ // do application analysis -
+ // use info.Data() to get the pointer for each element in blocksInfo
+
+ // After any desired analysis is finished, writer can now reuse data pointer
+
+Note, however, that the ``Get()`` method:
```

Comment:
When building these docs I get:

```
/Users/4nt/canonical_adios/docs/user_guide/source/engines/dataspaces.rst:55: WARNING: Unexpected indentation.
source/engines/inline.rst:28: WARNING: Error in "code-block" directive:
maximum 1 argument(s) allowed, 6 supplied.
```

Reply:
I fixed it with this diff

```
diff --git a/docs/user_guide/source/engines/inline.rst b/docs/user_guide/source/engines/inline.rst
index e4761151..4b86a93d 100644
--- a/docs/user_guide/source/engines/inline.rst
+++ b/docs/user_guide/source/engines/inline.rst
@@ -26,34 +26,35 @@ For successful operation, the writer will perform a step, then the reader will p
 Typical access pattern:
 
 .. code-block:: c++
- // ... Application data generation
 
- inlineWriter.BeginStep();
- inlineWriter.Put(var, data); // always use deferred mode
- inlineWriter.EndStep();
- // Unlike other engines, data should not be reused at this point,
- // though ADIOS cannot enforce this.
- // Must wait until reader is finished using the data.
+    // ... Application data generation
 
- inlineReader.BeginStep();
- auto blocksInfo = inlineReader.BlocksInfo(var, step);
- for (auto& info : blocksInfo)
- {
-     var.SetBlockSelection(info.BlockID);
-     inlineReader.Get(var, info);
- }
- inlineReader.EndStep();
+    inlineWriter.BeginStep();
+    inlineWriter.Put(var, data); // always use deferred mode
+    inlineWriter.EndStep();
+    // Unlike other engines, data should not be reused at this point,
+    // though ADIOS cannot enforce this.
+    // Must wait until reader is finished using the data.
 
- // do application analysis -
- // use info.Data() to get the pointer for each element in blocksInfo
+    inlineReader.BeginStep();
+    auto blocksInfo = inlineReader.BlocksInfo(var, step);
+    for (auto& info : blocksInfo)
+    {
+        var.SetBlockSelection(info.BlockID);
+        inlineReader.Get(var, info);
+    }
+    inlineReader.EndStep();
 
- // After any desired analysis is finished, writer can now reuse data pointer
+    // do application analysis -
+    // use info.Data() to get the pointer for each element in blocksInfo
+
+    // After any desired analysis is finished, writer can now reuse data pointer
 
 Note, however, that the ``Get()`` method:
 
 .. code-block:: c++
 
- void Engine::Get<T>(                                       \
+    void Engine::Get<T>(                                       \
         Variable<T>, typename Variable<T>::Info & info, const Mode);
 
 is preliminary, and may change in the future.
```

Reply:
Fixed. I think - I just tried to build the docs locally, but I think I'm having an issue with some of the python packages.